From 75892d97979f397a66730b97e8279941169e0316 Mon Sep 17 00:00:00 2001
From: YunQiang Su <syq@gcc.gnu.org>
Date: Fri, 23 Aug 2024 23:46:16 +0800
Subject: MIPS: Include missing mips16.S in libgcc/lib1funcs.S

mips16.S was missing since
commit 29b74545531f6afbee9fc38c267524326dbfbedf
Date:   Thu Jun 1 10:14:24 2023 +0800

    MIPS: Add speculation_barrier support

Without mips16.S included, some symbols will miss for mips16, and
so some software will fail to build.

libgcc/ChangeLog:

	* config/mips/lib1funcs.S: Includes mips16.S.

(cherry picked from commit 9522fc8bb7812f2ad50eb038e0938bfd958e730f)
---
 libgcc/config/mips/lib1funcs.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/libgcc/config/mips/lib1funcs.S
+++ b/libgcc/config/mips/lib1funcs.S
@@ -19,7 +19,7 @@ a copy of the GCC Runtime Library Except
 see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 <http://www.gnu.org/licenses/>.  */
 
-//#include "mips16.S"
+#include "mips16.S"
 
 #ifdef L_speculation_barrier
 
